[SendGrid Lists Destination] - doc for new SendGrid Lists Destination (#7301)
segmentio/segment-docs: src/connections/destinations/catalog/actions-sendgrid-audiences/index.md

> ---
> title: [SendGrid Lists (Actions)] Destination
> ---

Reviewer: The `engage: true` attribute adds the "This destination is only compatible with [Twilio Engage](https://segment.com/docs/engage)." to the destination dossier. For example, https://segment.com/docs/connections/destinations/catalog/actions-display-video-360/

Please update the `id` attribute when you have it. 

```suggestion
---
title: SendGrid Lists (Actions) Destination
engage: true
id: 
---
```

{% include content/plan-grid.md name="actions" %}

> [SendGrid Lists (Actions)](https://mc.sendgrid.com/contacts/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank”} help customers organize their email recipients into targeted groups, enabling them to send personalized, relevant content to specific audiences. This improves engagement, increases email deliverability, and streamlines campaign management.

Reviewer: ```suggestion
[SendGrid Lists (Actions)](https://mc.sendgrid.com/contacts/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank”} helps customers organize their email recipients into targeted groups, enabling them to send personalized, relevant content to specific audiences. This improves engagement, increases email deliverability, and streamlines campaign management.
```

> This destination is maintained by Segment. For any issues with the destination [contact the Segment Support team](mailto:friends@segment.com).

Reviewer: ```suggestion
This destination is maintained by Segment. For any issues with the destination, [contact the Segment Support team](mailto:friends@segment.com).
```

## Getting started

> **SendGrid Lists (Actions)** is designed to work with **Engage Audiences** only. The steps below outline how to create and connect the Destination to Engage and then to an Engage Audience. **SendGrid Lists (Actions)** is not designed to connect to regular Connections Sources. 

Reviewer: ```suggestion
SendGrid Lists (Actions) is designed to work with Engage Audiences only. The steps below outline how to create and connect the Destination to Engage and then to an Engage Audience. SendGrid Lists (Actions) is not designed to connect to Connections Sources. 
```

### Create a SendGrid API Key
1. Sign in to your SendGrid account, then navigate to **Settings** > **API Keys**.

> 2. Click **Create API Key** and follow the instructions to generate a new API key. Ensure Full Access permission is selected. 

Reviewer: ```suggestion
2. Click **Create API Key** and follow the instructions to generate a new API key. Be sure to grant the API key **Full Access** permission. 
```

> 3. Save the API key value securely, as you will need it in later steps.

Reviewer: ```suggestion
3. Save the API key value in a secure location, as you will need it in later steps.
```

> ### Create and connect 'SendGrid Lists (Actions)' to an Engage Space

Reviewer: ```suggestion
### Connect a SendGrid Lists (Actions) destination to an Engage Space
```

> 1. From your Segment workspace's home page, click on **Engage** > **Engage Settings** > **Settings**. Click the **Add destination** button. 

Reviewer: ```suggestion
1. From your Segment workspace's home page, navigate to **Engage** > **Engage Settings** > **Settings** and click **Add destination**. 
```

> 2. Search for **SendGrid Lists (Actions)** and select its tile. Click **Add Destination** and then **Confirm Source**.

Reviewer: ```suggestion
2. Search for SendGrid Lists (Actions) and select the SendGrid Lists (Action) tile. Click **Add Destination** and **Confirm Source**.
```

> 3. On the Basic Settings screen provide **Name** and **API Key** values in the specified fields. Enable the Destination. 

Reviewer: ```suggestion
3. On the Basic Settings screen, provide **Name** and **API Key** values in the specified fields, toggle "Enable destination" to on, and then click **Save Changes**. 
```

### Create a Mapping

> 1. From your Segment workspace's home page, click **Connections** > **Destinations**, then search for the **Destination** created in previous steps and click on its tile.

Reviewer: ```suggestion
1. From your Segment workspace's home page, click **Connections** > **Destinations** and select the SendGrid Lists (Actions) destination you previously created.
```

2. Click on **Mappings** > **New Mapping** > **Sync Audience** > **Save**. 
3. On the next screen, enable the Mapping using the **Status** toggle.

### Connect an Audience

> 1. From your Segment workspace's home page, navigate to the Audience you'd like to sync to SendGrid by clicking on **Engage** > **Audiences**, then selecting the Audience to connect. 

Reviewer: ```suggestion
1. From your Segment workspace's home page, navigate to **Engage** > **Audiences** and select the Audience you'd like to sync to SendGrid. 
```

> 2. Click **Add Destination**, then search for the **SendGrid Lists (Actions)** instance you created earlier and select its tile. 

Reviewer: ```suggestion
2. Click **Add Destination**, and select the SendGrid Lists (Actions) destination you previously created.
```

> 3. Enter a **List Name**, select **Default Setup**, and click **Save**. On the following screen, click **Add 1 Destination**. 

Reviewer: ```suggestion
3. Enter a **List Name**, select **Default Setup**, and click **Save**. When prompted, select **Add 1 Destination**. 
```

The **SendGrid Lists (Actions)** will now start to sync the **Engage Audience** to a **SendGrid List**.

{% include components/actions-fields.html %}

> ## Additional information

Author: hi @forstisabella yes absolutely! Do you want to do it or shall I?

### Does Segment create Lists in SendGrid?
Segment automatically creates Lists in SendGrid. The List will be named with the value provided in the **Name** field. If no name is provided in the **Name** field Segment will default to using the Audience's **Audience Key** value. 

### Does Segment create new Contacts in SendGrid?
Segment will create Contacts in SendGrid if a Contact doesn't already exist.

### Will Segment delete Contacts from SendGrid?

> Segment will not delete Contacts from SendGrid. If a user is removed from an Engage Audience, Segment will remove the Contact from the associated List in SendGrid but will not delete the Contact. 

Reviewer: ```suggestion
Segment doesn't delete Contacts from SendGrid. If you remove a user from an Engage Audience, Segment does remove the Contact from the associated SendGrid List, but doesn't delete the Contact from SendGrid. 
```

### Sending additional user traits
Segment supports sending Engage user profile traits to SendGrid Contact User Attributes. The following additional manual configuration steps are required: 

> 1. Use Engage's [Trait Enrichment](https://segment.com/docs/engage/trait-activation/trait-enrichment/) to configure Engage to include specific user profile traits when syncing users to the SendGrid List. 

Reviewer: ```suggestion
1. Use [Trait Enrichment](/docs/engage/trait-activation/trait-enrichment/) to include specific user profile traits when syncing users to a SendGrid List. 
```

2. Search for SendGrid Lists (Actions) and select the SendGrid Lists (Action) tile. Click **Add Destination** and **Confirm Source**.
3. On the Basic Settings screen, provide **Name** and **API Key** values in the specified fields, toggle "Enable destination" to on, and then click **Save Changes**. 

### Create a Mapping

1. From your Segment workspace's home page, click **Connections** > **Destinations** and select the SendGrid Lists (Actions) destination you previously created.
2. Click on **Mappings** > **New Mapping** > **Sync Audience** > **Save**. 
3. On the next screen, enable the Mapping using the **Status** toggle.

### Connect an Audience

1. From your Segment workspace's home page, navigate to **Engage** > **Audiences** and select the Audience you'd like to sync to SendGrid. 
2. Click **Add Destination**, and select the SendGrid Lists (Actions) destination you previously created.
3. Enter a **List Name**, select **Default Setup**, and click **Save**. When prompted, select **Add 1 Destination**. 

The SendGrid Lists (Actions) destination will now start to sync your Engage Audience to a SendGrid List.

{% include components/actions-fields.html %}

> ## Additional information

Reviewer: ```suggestion
## Troubleshooting
```

### Does Segment create Lists in SendGrid?
Segment automatically creates Lists in SendGrid. If you provide a value in the **Name** field, Segment names the List the value you provided. If you do not provide a name in the  **Name** field, Segment gives the List the Engage Audience's **Audience Key** value. 

### Does Segment create new Contacts in SendGrid?
Segment creates Contacts in SendGrid if a Contact doesn't already exist for the user.

### Does Segment delete Contacts from SendGrid?
Segment doesn't delete Contacts from SendGrid. If you remove a user from an Engage Audience, Segment does remove the Contact from the associated SendGrid List, but doesn't delete the Contact from SendGrid. 

> ### Sending additional user traits

Reviewer: ```suggestion
## Best practices

### Sending additional user traits
```

### Does Segment create new Contacts in SendGrid?
Segment creates Contacts in SendGrid if a Contact doesn't already exist for the user.

### Does Segment delete Contacts from SendGrid?
Segment doesn't delete Contacts from SendGrid. If you remove a user from an Engage Audience, Segment does remove the Contact from the associated SendGrid List, but doesn't delete the Contact from SendGrid. 

## Best practices

### Sending additional user traits
Segment supports sending Engage user profile traits to SendGrid Contact User Attributes. The following additional manual configuration steps are required: 

1. Use [Trait Enrichment](/docs/engage/trait-activation/trait-enrichment/) to include specific user profile traits when syncing users to a SendGrid List. 
2. Standard User Attributes: Use the [Sync Audience Action](#sync-audience-action)'s User Attributes field to map the following [Contact Profile Fields](https://www.twilio.com/docs/sendgrid/ui/managing-contacts/segmenting-your-contacts#contact-profile-fields){:target="_blank”} to SendGrid:
 - First Name
 - Last Name

>  - Phone Number (must be in [E.164](https://www.twilio.com/docs/glossary/what-e164) format)

Reviewer: ```suggestion
 - Phone Number (must be in [E.164](https://www.twilio.com/docs/glossary/what-e164){:target="_blank”} format)
```
